src/main/java/uk/ac/cam/cl/dtg/segue/api/QuestionFacade.java

```diff
@@ -232,6 +237,46 @@ public Response getQuestionsAnswered(@Context final HttpServletRequest request,
     }
   }
 
+  /**
+   * REST end point to provide five random questions.
+   *
+   * @param request  - this allows us to check to see if a user is currently logged in.
+   * @param subjects - a comma separated list of subjects
+   * @return a Response containing a gameboard object or containing a SegueErrorResponse.
+   */
+  @GET
+  @Path("/random")
+  @Produces(MediaType.APPLICATION_JSON)
+  @GZIP
+  public final Response getRandomQuestions(@Context final HttpServletRequest request,
+                                           @QueryParam("subjects") final String subjects) {
+    RegisteredUserDTO currentUser;
+
+    try {
+      currentUser = this.userManager.getCurrentRegisteredUser(request);
+
+      if (currentUser == null) {
```

Comment:
Can this actually be `null`? Despite what the Javadoc for `getCurrentRegisteredUser` says, it does not seem like the method returns `null`. Instead, it throws an exception when the user is not logged in.

Reply:
I was following the comment that I have just updated from the **getCurrentRegisteredUser** method in the UserAccountManager file. So that in future, no one working on with this method will be lead astray. The changes to this will be shown in my next commit.